instrumentation/active_record/lib/opentelemetry/instrumentation/active_record/instrumentation.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module ActiveRecord
+      # The Instrumentation class contains logic to detect and install the ActiveRecord instrumentation
+      class Instrumentation < OpenTelemetry::Instrumentation::Base
+        MINIMUM_VERSION = Gem::Version.new('5.2.0')
+
+        install do |_config|
+          require_dependencies
+          patch
+        end
+
+        present do
+          defined?(::ActiveRecord)
+        end
+
+        compatible do
+          gem_version >= MINIMUM_VERSION
+        end
+
+        private
+
+        def insert_class_methods_supported?
+          gem_version >= Gem::Version.new('6.0.0')
+        end
+
+        def gem_version
+          Gem.loaded_specs['activerecord'].version
+        end
+
+        def patch
+          ::ActiveRecord::Querying.prepend(Patches::Querying)
+          ::ActiveRecord::Persistence.prepend(Patches::Persistence)
+          ::ActiveRecord::Persistence::ClassMethods.prepend(Patches::PersistenceClassMethods)
+          ::ActiveRecord::Persistence::ClassMethods.prepend(Patches::PersistenceInsertClassMethods) if insert_class_methods_supported?
```

Comment:
Conditionally patching these methods as they were introduced in rails 6.0 https://edgeguides.rubyonrails.org/6_0_release_notes.html#active-record-notable-changes